### DISCUSSION TITLE
"Asset packs" for object replacement

### DISCUSSION THREAD

**Discussion by spacek531** [OpenRCT2/OpenRCT2#18704] [Feature requests]
This feature would allow users to override the object update tables at the following locations:

src/openrct2/park/Legacy.cpp
src/openrct2/rct1/Tables.cpp
src/openrct2/rct2/Rct2.cpp

And any others I have missed.

I hypothesize that a system similar to asset packs–or including this feature in asset packs–would be a simple way to manage the tables.

There are two ongoing projects which would benefit from user-defined update tables.

The RCT1 vehicles project would benefit from this. With user-defined object replacements, OpenRCT2 would not be required to host vehicles deemed unnecessary while allowing RCT1 enjoyers to import parks using the RCT1 objects.

The NEDesigns scenery update project would benefit from this. Multiple legacy custom scenery objects would be congregated into a single object that incorporates metadata fixes or graphical fixes. With user-defined object replacements, OpenRCT2 and the project can remain completely separate while allowing users to experience the benefits of the project.

**Comment by IntelOrca** [comment]
If the mappings were read from an external file rather than being hardcoded into the sources / executable, then that would make them a lot easier to edit.

**Comment by spacek531** [comment]
Does OpenRCT2 have—or have the capability for—mandatory asset packs or asset packs included in the distribution? I've not done anything with asset packs yet.

My expectation is it would be possible to create subsections in asset packs. One subsection would obviously be sprites, another sound files, and new subsection(s) would be for updating objects.

I expect the game can treat the OpenRCT2 default asset pack as always being the first asset pack in the list. Since asset packs have priority other earlier ones, having a default asset pack(s) will pose no problem for those who want to customize their experience with additional updates or changing the defaults.
